edx/edx-enterprise
# ENT-232: Add catalog and catalog courses endpoint to enterprise service

Hi @mattdrayer , @asadiqbal08 , @haikuginger ,

Kindly review this PR.

**Description:**
This PR adds api endpoints to fetch list of catalogs, catalog derails and catalog courses.

**JIRA:** https://openedx.atlassian.net/browse/ENT-232

**Testing instructions:**

1. Open enterprise api main page at '/enterprise/api/v1/' and make sure that a new api endpoint for catalogs is present (sample: "catalogs": "http://open.edx:8000/enterprise/api/v1/catalogs/").  
2. Open catalogs api endpoint at `enterprise/api/v1/catalogs/` and make sure it list all the catalogs accessible to the logged in user (All catalogs are accessible for staff users.).
3. Open catalog details page at `enterprise/api/v1/catalogs/1/`, and verify that catalog details are correct. Make sure `404` is returned if you try to access a catalog that does not exist.
4. Finally Open a catalog courses page at `enterprise/api/v1/catalogs/1/courses` and make sure all the courses are returned present in the selected catalog. Make sure pagination links are working.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
